### PR TITLE
Add support for DelegatedSending

### PR DESCRIPTION
**Pull request by Fang-**
Depends on urbit/azimuth#12, for which the `package.json` will need to be updated still.

Also, I saw [this](https://github.com/urbit/azimuth-js/blob/delegated-sending/contracts.js#L75-L79), and kind of just want to expose the `newEcliptic` etc functions, instead of the bare ABIs. Any objections, @jtobin?

### CODE REVIEW

File: check.js [hunk 1]
```diff
@@ -714,6 +715,49 @@ async function csrCanForfeit(contracts, batch, address) {
...
+ * @param {String} address - The caller address
+ * @return {Promise<Object>} A result and reason pair
+ */
+async function canSendInvitePoint(contracts, as, point, to, address) {
+  let res = await checkActivePointOwner(contracts, as, address);
+  if (!res.result) {
+    return res;
+  }
+  res.result = false;
+  if (to === address) {
```
Comment (jtobin) [inline check.js]: IIRC there's an `addressEquals` function in utils that makes sure these are checksummed before comparing them.  Worth using here?

File: delegatedSending.js [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+
+const internal = require('./internal/delegatedSending');
+
+/**
+ * Return the amount of invites left in the pool
+ * @param {Number} pool - Pool number
+ * @return {Promise<Number>} Number of invites remaining
+ */
+module.exports.pools = internal.pools
+module.exports.invitesInPool = internal.pools
```
Comment (jtobin) [inline delegatedSending.js:14]: What's the reason for having duplicate functions here (and below)?

File: internal/delegatedSending.js [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+
+module.exports.getPool = (contracts, point) => {
+  return contracts.delegatedSending.methods.getPool(point).call();
+}
+
+module.exports.canReceive = (contracts, recipient) => {
+  return contracts.delegatedSending.methods.canReceive(recipient).call();
+}
+
+// NB (jtobin):
```
Comment (jtobin) [inline internal/delegatedSending.js:28]: Note to self: we've got enough of these things now that I should look into this comment again at some point.. 😄

**Review by jtobin — COMMENTED** [review]
LGTM, a few minor comments here and there.

File: delegatedSending.js (hunk 2 above)
Comment (Fang-) [reply delegatedSending.js:14]: While `pools()` matches the contract's function name, it isn't exactly informative. Offering an alias here, `invitesInPool()`, should lead to more readable code down the line.

**Comment by Fang-** [comment]
Are these just npm shenanigans or am I doing something dumb here?

**Comment by jtobin** [comment]
I'll take a look.

**Comment by jtobin** [comment]
@Fang- 'twas just that the package-lock file needed to be updated. :+1:

File: package-lock.json [hunk 4]
```diff
@@ -236,7 +236,7 @@
     },
     "array-flatten": {
       "version": "1.1.1",
-      "resolved": "https://registry.npmjs.org/array-flatten/-/array-flatten-1.1.1.tgz",
+      "resolved": "http://registry.npmjs.org/array-flatten/-/array-flatten-1.1.1.tgz",
```
Comment (Fang-) [inline package-lock.json:239]: We went from https to http on this. npm why? ):

**Pull request merged**